lib/json.rb

```diff
@@ -2,88 +2,405 @@
+#   JSON.generate(Float(1.0)) # => "1.0"
+#
+# When the source is a Ruby \String, JSON.generate returns
+# a \String containing a \JSON string (with double-quotes):
+#   JSON.generate('A string') # => "\"A string\""
+#
+# When the source is +true+, +false+ or +nil+, JSON.generate returns
+# a \String containing the corresponding \JSON token:
+#   JSON.generate(true) # => "true"
+#   JSON.generate(false) # => "false"
+#   JSON.generate(nil) # => "null"
+#
+# When the source is none of the above, JSON.generate returns
+# a \String containing a \JSON string representation of the source:
+#   JSON.generate(:foo) # => "\"foo\""
+#   JSON.generate(Complex(0, 0)) # => "\"0+0i\""
+#   JSON.generate(Dir.new('.')) # => "\"#<Dir:0x0000000006bb30b8>\""
+#
+# == \JSON Additions
+#
+# When you "round trip" a non-\String object from Ruby to \JSON and back,
+# you have a new \String, instead of the object you began with:
+#   ruby0 = Range.new(0, 2)
+#   json = JSON.generate(ruby0)
+#   json # => "\"0..2\""
+#   ruby1 = JSON.parse(json)
+#   ruby1 # => "0..2"
+#   ruby1.class # => String
+#
+# You can use \JSON _additions_ to preserve the original object.
+# The addition is an extension of a ruby class, so that:
+# - \JSON.generate stores more information in the \JSON string.
+# - \JSON.parse, called with option +create_additions+,
+#   uses that information to create a proper Ruby object.
 #
-# <tt>JSON.generate</tt> only allows objects or arrays to be converted
-# to JSON syntax. <tt>to_json</tt>, however, accepts many Ruby classes
-# even though it acts only as a method for serialization:
+# This example shows a \Range being generated into \JSON
+# and parsed back into Ruby, both without and with
+# the addition for \Range:
+#   ruby = Range.new(0, 2)
+#   # This passage does not use the addition for Range.
+#   json0 = JSON.generate(ruby)
+#   ruby0 = JSON.parse(json0)
+#   # This passage uses the addition for Range.
+#   require 'json/add/range'
+#   json1 = JSON.generate(ruby)
+#   ruby1 = JSON.parse(json1, create_additions: true)
+#   # Make a nice display.
+#   display = <<EOT
+#   Generated JSON:
+#     Without addition:  #{json0} (#{json0.class})
+#     With addition:     #{json1} (#{json1.class})
+#   Parsed JSON:
+#     Without addition:  #{ruby0.inspect} (#{ruby0.class})
+#     With addition:     #{ruby1.inspect} (#{ruby1.class})
+#   EOT
+#   puts display
 #
+# This output shows the different results:
+#   Generated JSON:
+#     Without addition:  "0..2" (String)
+#     With addition:     {"json_class":"Range","a":[0,2,false]} (String)
+#   Parsed JSON:
+#     Without addition:  "0..2" (String)
+#     With addition:     0..2 (Range)
+#
+# The \JSON module includes additions for certain classes.
+# You can also craft custom additions.
+# See {Custom \JSON Additions}[#module-JSON-label-Custom+JSON+Additions].
+#
+# === Built-in Additions
+#
+# The \JSON module includes additions for certain classes.
+# To use an addition, +require+ its source:
```

Comment:
It would be nice to add an `*` after the classes included in `add/core` and put mention about `add/core` after the list

Reply:
Not sure what you mean, but again negative.  I've added a lot here, especially about additions.  Anyone who has more can create their own  PRs, right?

Reply:
Sure

Reply:
In the future, I recommend not marking points where you disagree with the reviewer as "resolved" until (s)he has agreed

Reply:
Makes sense.